addons/sale_order_dates/models/sale_order.py

```diff
@@ -24,30 +24,28 @@ class SaleOrder(models.Model):
                                           "processed as soon as possible. In that case the expected "
                                           "date will be computed using the default method: based on "
                                           "the Product Lead Times and the Company's Security Delay.")
-    effective_date = fields.Date(compute='_compute_picking_ids', string='Effective Date', store=True,
-                                 help="Date on which the first Delivery Order was created.")
+    effective_date = fields.Date(compute='_compute_effective_date', string='Effective Date', store=True,
+                                 help="Date on which the first shipment was successfully delivered to customer.")
 
-    @api.depends('date_order', 'order_line.customer_lead')
+    @api.depends('order_line.customer_lead', 'confirmation_date', 'picking_policy')
     def _compute_commitment_date(self):
         """Compute the commitment date"""
         for order in self:
             dates_list = []
-            order_datetime = fields.Datetime.from_string(order.date_order)
+            confirm_date = fields.Datetime.from_string(order.confirmation_date or fields.Datetime.now())
```

Comment:
Shouldn't it be based on create_date is no confirmation date ? Not sure, just asking.